### DISCUSSION TITLE
Create File from Template: Ask for Filename, then set both the name and the first Heading as such

### DISCUSSION THREAD

**Discussion by Gewerd-Strauss** [SilentVoid13/Templater#541] [Help]
Hello,

I have a general template structure I want to create. On Execution, I want to

1. Be asked what to name the file.
2. Set the notes filename to my input
3. Insert the rest of the template and populate it accordingly

The rudimentary structure of the end product, upon finished template-execution is as follows:

```
---
Name: FileName
Tags: \#Tag-One-Referencing-Section\#Tag-Two-Referencing-Chapter,#Tag-Three-Name-Of-Book
---

# FileName

and everything below can be handled as separate templates.

## 
```

Example for `Dwarf.md`:

```
---
Name: Dwarf
Tags: #Part-1:-Creating-a-Character #Chapter-2:-Races #Player's Handbook
---

# Dwarf

##
```

Now. The problem I have is that

1. I don't understand neither how to, nor if I am supposed to use `tp.file.create_new()` for this.
As I understand it, I can create a file from a template initially, instead of just inserting them post creation. I suspect that's what I am looking for, but I am not sure.

The template folder is located at `VaultRoot/002 templates`
My very rudimentary first shot, which doesn't work, was this:

File 1: 

Book_CreateChapter ← Template, placed in template folder
```
<% tp.file.create_new([[002 templates/DND_BookEntry_Template]],tp.system.prompt("Set Filename","",1),true) %>
```

File 2: 

DND_BookEntry_Template ← template, also in template-folder

``` 
---
Name:<% tp.file.title %>
Tags: #PartName, #ChapterName, #SourcebookName1 #SourcebookName2

---

# <% tp.file.title %>
``` 

Every Note also contains three tags, all of which should be chooseable by user at creation as well. However, as I _suspect_ that to just be a permutation of what I suspect I must do for the filename fed into `create_new()`, I only mention this. 

---

What exactly am I missing here? I am not exactly experienced in the arts of this plugin, but from what I see other people do with it I am both eager to learn, (kinda jealous that I suck at it), and utterly bewildered by my apparent lack of insight here :/

**Comment by AB1908** [comment]
`tp.file.create_new()` expects a `TFile` if you want to use a template. Can get `TFile`s by using `tp.file.find_tfile()` or with `app.vault.getAbstractFileByPath()`. Pass that as an argument.

The `prompt` and `suggester` need to be `await`ed.

```js
const templateTFile = tp.file.find_tfile("[[template]]");
const fileName = await tp.system.prompt("Enter file name");
await tp.file.create_new(templateTFile, fileName, false,);
```
